packages/client-readmodel-writer-sql/src/readModelService.ts

```diff
@@ -0,0 +1,215 @@
+import {
+  Client,
+  ClientId,
+  dateToString,
+  Key,
+  PurposeId,
+  UserId,
+} from "pagopa-interop-models";
+import { ClientReadModelService } from "pagopa-interop-readmodel";
+import {
+  DrizzleTransactionType,
+  clientInReadmodelClient,
+  clientUserInReadmodelClient,
+  clientPurposeInReadmodelClient,
+  clientKeyInReadmodelClient,
+  ClientUserSQL,
+  ClientPurposeSQL,
+  ClientKeySQL,
+  DrizzleReturnType,
+} from "pagopa-interop-readmodel-models";
+import { and, eq, lte } from "drizzle-orm";
+
+// eslint-disable-next-line @typescript-eslint/explicit-function-return-type
+export function readModelServiceBuilder(
+  db: DrizzleReturnType,
+  clientReadModelService: ClientReadModelService
+) {
+  const updateMetadataVersionInClientTables = async (
+    tx: DrizzleTransactionType,
+    clientId: ClientId,
+    newMetadataVersion: number
+  ): Promise<void> => {
+    const clientRelatedTables = [
+      clientInReadmodelClient,
+      clientUserInReadmodelClient,
+      clientPurposeInReadmodelClient,
+      clientKeyInReadmodelClient,
+    ];
+
+    for (const table of clientRelatedTables) {
+      await tx
+        .update(table)
+        .set({ metadataVersion: newMetadataVersion })
+        .where(
+          and(
+            eq("clientId" in table ? table.clientId : table.id, clientId),
+            lte(table.metadataVersion, newMetadataVersion)
+          )
+        );
+    }
+  };
```

Comment:
I was thinking that maybe also this `updateMetadataVersionInXXXTables` could be generalized as we did with `checkMetadataVersion`.
Not in these PRs, but maybe after you have merged all the readmodel-writer-sql services? If you agree you could track it on Jira as part of the feature